react/components/SearchQuery.js

```diff
@@ -50,22 +57,52 @@ const useCombinedRefetch = (productRefetch, facetsRefetch) => {
   )
 }
 
-const useCorrectPage = ({ page, query, map, orderBy }) => {
-  const pageRef = useRef(page)
+const isCurrentDifferent = (ref, currentVal) => ref.current !== currentVal
+
+const useShouldResetPage = (query, map, orderBy) => {
   const queryRef = useRef(query)
```

Comment:
I think this might be a moment to refactor all of these refs to use a reducer state model.

Then you can just dispatch the reset action when needed. It may be done in a later PR though.

Reply:
Nice suggestion!
I have just created an issue about that.
https://github.com/vtex-apps/search-result/issues/330

I could make this PR by myself when I have a chance.